storagemarket/impl/provider_environments_test.go

```diff
@@ -0,0 +1,50 @@
+package storageimpl
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/ipfs/go-cid"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/go-fil-markets/shared_testutil"
+)
+
+func TestGeneratePieceCommitment(t *testing.T) {
+	// both payload.txt and payload2.txt are about 18kb long
+	pieceSize := abi.PaddedPieceSize(32768)
+	_, carV2File1 := shared_testutil.CreateDenseCARv2(t, filepath.Join("storagemarket", "fixtures", "payload.txt"))
+	defer os.Remove(carV2File1)
+	_, carV2File2 := shared_testutil.CreateDenseCARv2(t, filepath.Join("storagemarket", "fixtures", "payload2.txt"))
+	defer os.Remove(carV2File2)
+
+	commP1 := genProviderCommP(t, carV2File1, pieceSize)
+	commP2 := genProviderCommP(t, carV2File2, pieceSize)
+
+	commP3 := genProviderCommP(t, carV2File1, pieceSize)
+	commP4 := genProviderCommP(t, carV2File2, pieceSize)
+
+	require.Equal(t, commP1, commP3)
```

Comment:
I do not understand the value of this test -- are we surprised that to runs of the same function on the same file generate the same output while two runs different files produce different output? That seems somewhat self-evident. If the goal is to test that is actually produces "the right output" I feel like this test is incomplete. However, I also feed the loginc in the code itself is small enough that I'm not sure why we need to test this, especially since it primarily relies on code from commp-utils

Reply:
Good question!
Will follow up with @aarshkshah1992